tpl: TestMethodToName fails with gccgo

With gccgo-7:

```
$ go-7 version
go version go1.8.1 gccgo (Debian 7.1.0-10) 7.1.0 linux/amd64
```

`go-7 test` produces the following error:

```
=== RUN   TestMethodToName
--- FAIL: TestMethodToName (0.00s)
        Error Trace:    templatefuncRegistry_test.go:32
	Error:		Not equal: "MyTestMethod" (expected)
			        != "$thunk0" (actual)
		
FAIL
FAIL	github.com/gohugoio/hugo/tpl/internal	0.916s
```

Is it safe to just skip the test on gccgo with `// +build !gccgo` in tpl/internal/templatefuncRegistry_test.go (which contains only one test, TestMethodToName())?

Or is this a concern that may cause a gccgo-compiled Hugo to fail with certain templates?

Many thanks!

References:

* https://github.com/golang/go/issues/3901
* https://bugs.launchpad.net/gocheck/+bug/1250253

--- a/tpl/internal/templatefuncRegistry_test.go
+++ b/tpl/internal/templatefuncRegistry_test.go
@@ -14,6 +14,7 @@
 package internal
 
 import (
+	"runtime"
 	"testing"
 
 	"github.com/stretchr/testify/require"
@@ -29,5 +30,9 @@ func (t *Test) MyTestMethod() string {
 func TestMethodToName(t *testing.T) {
 	test := &Test{}
 
-	require.Equal(t, "MyTestMethod", methodToName(test.MyTestMethod))
+	if runtime.Compiler == "gccgo" {
+		require.Equal(t, "$thunk0", methodToName(test.MyTestMethod))
+	} else {
+		require.Equal(t, "MyTestMethod", methodToName(test.MyTestMethod))
+	}
 }

